privacyidea/lib/crypto.py

```diff
@@ -403,25 +413,34 @@ def aes_encrypt_b64(key, data):
     :return: base64 encrypted output, containing IV and encrypted data
     :rtype: str
     """
+    # pad data
+    block_size = algorithms.AES.block_size // 8
+    num_pad = block_size - (len(data) % block_size)
+    data = data + six.int2byte(num_pad) * num_pad
     iv = geturandom(16)
-    encdata = aes_encrypt(key, iv, data)
+    encdata = aes_cbc_encrypt(key, iv, data)
     return b64encode_and_unicode(iv + encdata)
 
 
-def aes_decrypt_b64(key, data_b64):
+def aes_decrypt_b64(key, enc_data_b64):
     """
     This function decrypts base64 encoded data (containing the IV)
     using AES-128-CBC. Used for PSKC
 
     :param key: binary key
-    :param data_b64: base64 encoded data (IV + encdata)
-    :type data_b64: str
+    :param enc_data_b64: base64 encoded data (IV + encdata)
+    :type enc_data_b64: str
     :return: encrypted data
     """
-    data_bin = base64.b64decode(data_b64)
+    data_bin = base64.b64decode(enc_data_b64)
     iv = data_bin[:16]
     encdata = data_bin[16:]
-    output = aes_decrypt(key, iv, encdata)
+    output = aes_cbc_decrypt(key, iv, encdata)
+
+    # remove padding
+    padding = six.indexbytes(output, len(output) - 1)
+    output = output[0:-padding]
```

Comment:
same as above: could we use the cryptography unpadders?

Reply:
will do.

Reply:
fixed in f2e066a6